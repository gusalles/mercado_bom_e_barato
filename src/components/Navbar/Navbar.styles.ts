import { Container, ContainerProps, styled } from '@mui/material';

export const NavBarContainer = styled(Container)<ContainerProps>(
  ({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderRadius: '32px',
    backgroundColor: theme.palette.background.paper,
    padding: '16px 0',
    marginTop: '40px',
    [theme.breakpoints.down('md')]: {
      width: 'auto',
      padding: '18px 0',
      margin: '0 16px',
      backgroundColor: theme.palette.background.default,
    },
  })
);

export const NavBarSection = styled('section')(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  gap: '2rem',
  [theme.breakpoints.down('md')]: {
    gap: '1.125rem',
  },
}));
