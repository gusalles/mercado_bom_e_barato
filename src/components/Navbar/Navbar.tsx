import { Box } from '@mui/material';
import Image from 'next/image';
import { Logo } from '../Logo';

export function Navbar() {
  return (
    <Box
      component="nav"
      sx={{
        borderRadius: '32px',
        bgcolor: { sm: 'transparent', md: 'background.paper' },
        paddingY: { sm: '8px', md: '16px' },
        paddingX: { sm: '8px', md: '36px' },
        marginTop: { sm: '0', md: '40px' },
        marginX: { sm: '0', md: '96px' },
      }}
    >
      <Logo />
    </Box>
  );
}
