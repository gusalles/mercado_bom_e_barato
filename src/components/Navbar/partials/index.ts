export * from './BackButton';
export * from './NavbarTitle';
export * from './NavbarLink';
