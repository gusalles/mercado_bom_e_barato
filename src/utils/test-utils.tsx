import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

export * from '@testing-library/react';

export function renderWithClient(ui: ReactElement) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
  );
}
