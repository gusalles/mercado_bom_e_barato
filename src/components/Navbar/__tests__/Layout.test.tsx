import { render, screen, userEvent } from '@/utils/test-utils';
import { Navbar } from '../Navbar';

import routerMock from 'next-router-mock';

describe('Navbar', () => {
  it('should render', () => {
    render(<Navbar />);

    const navigation = screen.getByRole('navigation');

    expect(navigation).toMatchSnapshot();
  });

  it('should redirect to home when click in Home anchor', () => {
    render(<Navbar />);

    const link = screen.getByRole('link');

    userEvent.click(link);

    expect(routerMock.asPath).toEqual('/');
  });
});
