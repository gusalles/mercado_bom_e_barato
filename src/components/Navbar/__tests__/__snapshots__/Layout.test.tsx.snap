// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should render 1`] = `
<nav
  class="MuiContainer-root MuiContainer-maxWidthLg css-14qi191-MuiContainer-root"
>
  <section
    class="css-1lkr33r"
  >
    <img
      alt="Logo Mercado B&B"
      data-nimg="1"
      decoding="async"
      height="40"
      src="/logo.svg"
      style="color: transparent;"
      width="145"
    />
  </section>
  <section
    class="css-1lkr33r"
  >
    <a
      class="css-j5ksg0"
      href="/"
    >
      In√≠cio
    </a>
  </section>
  <section
    class="css-1lkr33r"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-25to7p-MuiSvgIcon-root"
      data-testid="PersonOutlineOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4m0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-25to7p-MuiSvgIcon-root"
      data-testid="ShoppingCartOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2zM6.16 6h12.15l-2.76 5H8.53zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
      />
    </svg>
  </section>
</nav>
`;
